src/main/java/uk/gov/digital/ho/pttg/application/namematching/NameMatchingCandidatesService.java

```diff
@@ -3,59 +3,33 @@
 
 @Service
 public class NameMatchingCandidatesService {
 
-    private NameMatchingCandidateGenerator nameCombinations;
-    private NameMatchingCandidateGenerator multipleLastNames;
-    private NameMatchingCandidateGenerator specialCharacters;
-    private NameMatchingCandidateGenerator aliasCombinations;
-    private NameMatchingCandidateGenerator entireNonAliasName;
-    private NameMatchingCandidateGenerator entireLastNameAndEachFirstName;
-    private NameMatchingCandidateGenerator namesWithFullStopSpaceCombinations;
-
-    public NameMatchingCandidatesService(NameMatchingCandidateGenerator nameCombinations,
-                                         NameMatchingCandidateGenerator multipleLastNames,
-                                         NameMatchingCandidateGenerator specialCharacters,
-                                         NameMatchingCandidateGenerator aliasCombinations,
-                                         NameMatchingCandidateGenerator entireNonAliasName,
-                                         NameMatchingCandidateGenerator entireLastNameAndEachFirstName,
-                                         NameMatchingCandidateGenerator namesWithFullStopSpaceCombinations) {
-        this.nameCombinations = nameCombinations;
-        this.multipleLastNames = multipleLastNames;
-        this.specialCharacters = specialCharacters;
-        this.aliasCombinations = aliasCombinations;
-        this.entireNonAliasName = entireNonAliasName;
-        this.entireLastNameAndEachFirstName = entireLastNameAndEachFirstName;
-        this.namesWithFullStopSpaceCombinations = namesWithFullStopSpaceCombinations;
+    private List<NameMatchingCandidateGenerator> candidateStrategies;
+
+    public NameMatchingCandidatesService(List<NameMatchingCandidateGenerator> candidateStrategies) {
+        this.candidateStrategies = candidateStrategies;
     }
 
     public List<CandidateName> generateCandidateNames(String firstNames, String lastNames, String aliasSurnames) {
-        InputNames inputNames = new InputNames(firstNames, lastNames, aliasSurnames);
-
-        List<CandidateName> candidates = new ArrayList<>(entireNonAliasName.generateCandidates(inputNames));
-
-        candidates.addAll(entireLastNameAndEachFirstName.generateCandidates(inputNames));
 
-        candidates.addAll(multipleLastNames.generateCandidates(inputNames));
-
-        candidates.addAll(namesWithFullStopSpaceCombinations.generateCandidates(inputNames));
-
-        if (inputNames.hasAliasSurnames()) {
-            candidates.addAll(aliasCombinations.generateCandidates(inputNames));
-        } else {
-            candidates.addAll(nameCombinations.generateCandidates(inputNames));
-        }
+        InputNames inputNames = new InputNames(firstNames, lastNames, aliasSurnames);
 
-        candidates.addAll(specialCharacters.generateCandidates(inputNames));
+        List<CandidateName> candidates = candidateStrategies
```

Comment:
We call the interface Generators not Strategies and we should be consistent. But I actually prefer Strategies as a name so maybe we can change it at some point?